src/internal/at.rs

```diff
@@ -822,30 +818,17 @@ pub fn subtable_data(b: &Bytes, offset: u32, kind: LookupKind, fmt: u16) -> Opti
     }
 }
 
-pub fn validate_coverage(b: &Bytes, coverage_offset: u32) -> Option<()> {
+fn validate_coverage(b: &Bytes, coverage_offset: u32) -> bool {
     if coverage_offset == 0 {
-        return None;
+        return false;
     }
     let base = coverage_offset as usize;
-    let fmt = b.read::<u16>(base)?;
-    let len = b.read::<u16>(base + 2)? as usize;
     let arr = base + 4;
-    match fmt {
-        1 => {
-            if !b.check_range(arr, len * 2) {
-                None
-            } else {
-                Some(())
-            }
-        }
-        2 => {
-            if !b.check_range(arr, len * 6) {
-                None
-            } else {
-                Some(())
-            }
-        }
-        _ => None,
+    match (b.read::<u16>(base), b.read::<u16>(base + 2)) {
+        (Some(_), Some(0)) => false,
+        (Some(1), Some(len)) => !!b.check_range(arr, len as usize * 2),
+        (Some(2), Some(len)) => !!b.check_range(arr, len as usize * 6),
```

Comment:
The double negations (`!!`) could be removed I think

Reply:
No, check_range returns a nonempty option, this coerces it to a bool.